test/e2e/harness.go

```diff
@@ -73,18 +73,18 @@ func newHarness() *harness {
 		leaderKeyPair := keys.Ed25519KeyPairForTests(0)
 		for i := 0; i < LOCAL_NETWORK_SIZE; i++ {
 			nodeKeyPair := keys.Ed25519KeyPairForTests(i)
-			node := bootstrap.NewNode(
-				fmt.Sprintf(":%d", 8080+i),
-				nodeKeyPair.PublicKey(),
-				nodeKeyPair.PrivateKey(),
+
+			cfg := config.ForProduction().OverrideNodeSpecificValues(
 				federationNodes,
 				gossipPeers,
 				uint16(firstRandomPort+i),
+				nodeKeyPair.PublicKey(),
+				nodeKeyPair.PrivateKey(),
 				leaderKeyPair.PublicKey(),
-				consensus.CONSENSUS_ALGO_TYPE_BENCHMARK_CONSENSUS,
-				logger,
-				processorArtifactPath,
-			)
+				consensus.CONSENSUS_ALGO_TYPE_BENCHMARK_CONSENSUS)
+			cfg.SetString(config.PROCESSOR_ARTIFACT_PATH, processorArtifactPath)
```

Comment:
I don't like this. It should be part of the factory method somehow. Please try to find a better way, if we leave it here it's broken windows syndrome.